docs/docs/jsx-in-depth.md

```diff
@@ -0,0 +1,312 @@
+
+For example, this code will not run as expected:
+
+```js
+import React from 'react';
+
+function hello(props) {
+  // This use of <div> is legitimate because div is a valid HTML tag
+  return <div>Hello {props.toWhat}</div>;
+}
+
+export default function HelloWorld() {
+  // This code attempts to create an HTML <hello> tag and fails
+  return <hello toWhat="World" />
+}
+```
+
+You cannot use a general expression as the React element type. If you do want to use a general expression to indicate the type of the element, just assign it to a capitalized variable first. For example, if you have a `findComponent()` function that returns a component, and you want to render one:
+
+```js
+import React from 'react';
+import findComponent from './findComponent';
+
+function render1() {
+  // Not valid JSX
+  return <findComponent() />;
+}
+
+function render2() {
+  var MyComponent = findComponent();
+
+  // Valid JSX
+  return <MyComponent />;
+}
+```
+
+## Props in JSX
+
+There are several different ways to specify props in JSX.
+
+### JavaScript Expressions
+
+You can pass any JavaScript expression as a prop, by surrounding it with `{}`. For example, in this JSX:
+
+```js
+<MyComponent foo={1 + 2 + 3 + 4} />
+```
+
+For `MyComponent`, The value of `props.foo` will be `10` because the expression `1 + 2 + 3 + 4` gets evaluated.
+
+`if` statements and `for` loops are not expressions in JavaScript, so they can't be used in JSX directly. Instead, you can put these in the surrounding code. For example:
+
+```js
+function NumberDescriber(props) {
+  var description;
+  if (props.number % 2 == 0) {
+    description = 'even';
+  } else {
+    description = 'odd';
+  }
+  return <div>{props.number} is an {description} number</div>;
+}
+```
+
+### String Literals
+
+You can pass a string literal as a prop. These two JSX expressions are equivalent:
+
+```js
+<MyComponent message="hello world" />
+
+<MyComponent message={"hello world"} />
+```
+
+When you pass a string literal, its value is HTML-unescaped. So these two JSX expressions are equivalent:
+
+```js
+<MyComponent message="&lt;3" />
+
+<MyComponent message={"<3"} />
+```
+
+This behavior is usually not relevant. It's useful for `children`, but not for most props. It's only mentioned here for completeness.
+
+### Props Default to "True"
+
+If you pass no value for a prop, it defaults to `true`. These two JSX expressions are equivalent:
+
+```js
+<MyTextBox autocomplete />
+
+<MyTextBox autocomplete={true} />
+```
+
+In general, we don't recommend using this. It's more consistent to just use the second form and explicitly pass `true`. This behavior is just there so that it matches the behavior of HTML.
```

Comment:
In general, we don't recommend using this form because it can be confused with the ES6 object shorthand `{autocomplete}` which is short for `{autocomplete: autocomplete}` rather than `{autocomplete: true}`.